diff --git a/pkgs/applications/editors/neovim/utils.nix b/pkgs/applications/editors/neovim/utils.nix
index cb0c005759b..1df6f7250aa 100644
--- a/pkgs/applications/editors/neovim/utils.nix
+++ b/pkgs/applications/editors/neovim/utils.nix
@@ -102,10 +102,13 @@ let
           # as expected by packdir
           packDirArgs.myNeovimPackages = myVimPackage;

+          packDir = vimUtils.packDir packDirArgs;
+
           # vim accepts a limited number of commands so we join them all
           flags = [
             "--cmd" (lib.intersperse "|" hostProviderViml)
-            "--cmd" "set packpath^=${vimUtils.packDir packDirArgs}"
+            "--cmd" "set packpath^=${packDir}"
+            "--cmd" "set rtp^=${packDir}"
             ];
         in
         [
