diff --git a/pkgs/by-name/ni/nix-plugin-pijul/package.nix b/pkgs/by-name/ni/nix-plugin-pijul/package.nix
index bd83cecddb..c39ef22811 100644
--- a/pkgs/by-name/ni/nix-plugin-pijul/package.nix
+++ b/pkgs/by-name/ni/nix-plugin-pijul/package.nix
@@ -81,6 +81,7 @@ stdenv.mkDerivation (finalAttrs: {
     {
       stable = localRepoCheck nixVersions.stable;
       latest = localRepoCheck nixVersions.latest;
+      git = localRepoCheck nixVersions.git;
       nix_2_24 = localRepoCheck nixVersions.nix_2_24;
     };
 
diff --git a/pkgs/tools/package-management/nix/default.nix b/pkgs/tools/package-management/nix/default.nix
index 8fab66bb2a..9663f5bfd4 100644
--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -260,6 +260,18 @@ lib.makeExtensible (
       # depend on the components they need in `nixComponents_2_26`.
       nix_2_26 = addTests "nix_2_26" self.nixComponents_2_26.nix-everything;
 
+      git = common rec {
+        version = "2.25.0";
+        suffix = "pre20241101_${lib.substring 0 8 src.rev}";
+        src = fetchFromGitHub {
+          owner = "NixOS";
+          repo = "nix";
+          rev = "2e5759e3778c460efc5f7cfc4cb0b84827b5ffbe";
+          hash = "sha256-E1Sp0JHtbD1CaGO3UbBH6QajCtOGqcrVfPSKL0n63yo=";
+        };
+        self_attribute_name = "git";
+      };
+
       latest = self.nix_2_26;
 
       # The minimum Nix version supported by Nixpkgs
@@ -293,8 +305,7 @@ lib.makeExtensible (
         ) (lib.range 4 23)
       )
       // {
-        unstable = throw "nixVersions.unstable has been removed. use nixVersions.latest or the nix flake.";
-        git = throw "nixVersions.git has been removed. use nixVersions.latest or the nix flake.";
+        unstable = throw "nixVersions.unstable has been removed. use nixVersions.git or the nix flake.";
       }
     )
   )
diff --git a/pkgs/tools/package-management/nix/update-all.sh b/pkgs/tools/package-management/nix/update-all.sh
index 547721d8b3..a2b459e67e 100755
--- a/pkgs/tools/package-management/nix/update-all.sh
+++ b/pkgs/tools/package-management/nix/update-all.sh
@@ -36,3 +36,9 @@ for name in $nix_versions; do
         break
     fi
 done
+
+commit_json=$(curl -s https://api.github.com/repos/NixOS/nix/commits/master) # format: 2024-11-01T10:18:53Z
+date_of_commit=$(echo "$commit_json" | jq -r '.commit.author.date')
+suffix="pre$(date -d "$date_of_commit" +%Y%m%d)_"
+sed -i -e "s|\"pre[0-9]\{8\}_|\"$suffix|g" "$SCRIPT_DIR/default.nix"
+nix-update --override-filename "$SCRIPT_DIR/default.nix" --version branch --build --commit "nixVersions.git"
diff --git a/pkgs/top-level/release-cross.nix b/pkgs/top-level/release-cross.nix
index ad45dcedb1..ded7924f14 100644
--- a/pkgs/top-level/release-cross.nix
+++ b/pkgs/top-level/release-cross.nix
@@ -70,6 +70,7 @@ let
     gmp = nativePlatforms;
     libcCross = nativePlatforms;
     nix = nativePlatforms;
+    nixVersions.git = nativePlatforms;
     mesa = nativePlatforms;
     rustc = nativePlatforms;
     cargo = nativePlatforms;
