From f5772f996fdc535adcae5ff475497d400f37909f Mon Sep 17 00:00:00 2001
From: Kirill Kuznetsov <kdevlab@yandex.ru>
Date: Tue, 15 Apr 2025 14:07:43 +0300
Subject: [PATCH] Remove mesa drivers

---
 pkgs/development/libraries/mesa/default.nix | 3 ---
 1 files changed, 3 deletions(-)

diff --git a/pkgs/development/libraries/mesa/default.nix b/pkgs/development/libraries/mesa/default.nix
index 255cda5a742372..07fca3455f6665 100644
--- a/pkgs/development/libraries/mesa/default.nix
+++ b/pkgs/development/libraries/mesa/default.nix
@@ -397,9 +397,6 @@ stdenv.mkDerivation {
     inherit (libglvnd) driverLink;
     inherit llvmPackages;
 
-    # for compatibility
-    drivers = lib.warn "`mesa.drivers` is deprecated, use `mesa` instead" mesa;
-
     tests.outDoesNotDependOnLLVM = stdenv.mkDerivation {
       name = "mesa-does-not-depend-on-llvm";
       buildCommand = ''
