diff --git a/nixos/tests/kernel-generic.nix b/nixos/tests/kernel-generic.nix
index 7ee734a1eff..a162888e1fe 100644
--- a/nixos/tests/kernel-generic.nix
+++ b/nixos/tests/kernel-generic.nix
@@ -30,6 +30,7 @@ let
       linux_5_4_hardened
       linux_5_10_hardened
       linux_5_15_hardened
+      linux_5_19_hardened
       linux_6_0_hardened
 
       linux_testing;
diff --git a/pkgs/os-specific/linux/kernel/hardened/patches.json b/pkgs/os-specific/linux/kernel/hardened/patches.json
index bfcfa874add..713b566439e 100644
--- a/pkgs/os-specific/linux/kernel/hardened/patches.json
+++ b/pkgs/os-specific/linux/kernel/hardened/patches.json
@@ -39,6 +39,16 @@
         "sha256": "16d4d4g5n2g6jpp8bvad1bm1l0b0nb4ckwsmq6w2g3538xrrzf8d",
         "version": "5.15.78"
     },
+    "5.19": {
+        "patch": {
+            "extra": "-hardened1",
+            "name": "linux-hardened-5.19.17-hardened1.patch",
+            "sha256": "0fs9zdwxbcc3fi4w1qylvvv387ih3g0dnagw5pvq3r2k2g5fflya",
+            "url": "https://github.com/anthraxx/linux-hardened/releases/download/5.19.17-hardened1/linux-hardened-5.19.17-hardened1.patch"
+        },
+        "sha256": "12cly10lad12idjwlgh2g0pp4hhj57h2qi4fy6jg1lbsm62b6fy9",
+        "version": "5.19.17"
+    },
     "5.4": {
         "patch": {
             "extra": "-hardened1",
diff --git a/pkgs/os-specific/linux/kernel/linux-5.19.nix b/pkgs/os-specific/linux/kernel/linux-5.19.nix
new file mode 100644
index 00000000000..6137c36cb5c
--- /dev/null
+++ b/pkgs/os-specific/linux/kernel/linux-5.19.nix
@@ -0,0 +1,18 @@
+{ lib, buildPackages, fetchurl, perl, buildLinux, nixosTests, modDirVersionArg ? null, ... } @ args:
+
+with lib;
+
+buildLinux (args // rec {
+  version = "5.19.17";
+
+  # modDirVersion needs to be x.y.z, will automatically add .0 if needed
+  modDirVersion = if (modDirVersionArg == null) then concatStringsSep "." (take 3 (splitVersion "${version}.0")) else modDirVersionArg;
+
+  # branchVersion needs to be x.y
+  extraMeta.branch = versions.majorMinor version;
+
+  src = fetchurl {
+    url = "mirror://kernel/linux/kernel/v5.x/linux-${version}.tar.xz";
+    sha256 = "12cly10lad12idjwlgh2g0pp4hhj57h2qi4fy6jg1lbsm62b6fy9";
+  };
+} // (args.argsOverride or { }))
diff --git a/pkgs/os-specific/linux/zfs/default.nix b/pkgs/os-specific/linux/zfs/default.nix
index 5309c6abe24..00afcfbda93 100644
--- a/pkgs/os-specific/linux/zfs/default.nix
+++ b/pkgs/os-specific/linux/zfs/default.nix
@@ -16,7 +16,7 @@
 , enablePython ? true
 
 # for determining the latest compatible linuxPackages
-, linuxPackages_5_15 ? pkgs.linuxKernel.packages.linux_5_15
+, linuxPackages_5_19 ? pkgs.linuxKernel.packages.linux_5_19
 , linuxPackages_6_0 ? pkgs.linuxKernel.packages.linux_6_0
 }:
 
@@ -218,7 +218,7 @@ in {
   zfsStable = common {
     # check the release notes for compatible kernels
     kernelCompatible = kernel.kernelOlder "5.20";
-    latestCompatibleLinuxPackages = linuxPackages_5_15;
+    latestCompatibleLinuxPackages = linuxPackages_5_19;
 
     # this package should point to the latest release.
     version = "2.1.6";
diff --git a/patches/revert-kernel-drop-5.19.patch b/patches/revert-kernel-drop-5.19.patch
index 50a5fd0a82b..623bc6a4cd7 100644
--- a/patches/revert-kernel-drop-5.19.patch
+++ b/patches/revert-kernel-drop-5.19.patch
@@ -77,48 +77,3 @@ index 5309c6abe24..00afcfbda93 100644
  
      # this package should point to the latest release.
      version = "2.1.6";
-diff --git a/pkgs/top-level/linux-kernels.nix b/pkgs/top-level/linux-kernels.nix
-index efd71e23ce7..71c5750c75a 100644
---- a/pkgs/top-level/linux-kernels.nix
-+++ b/pkgs/top-level/linux-kernels.nix
-@@ -166,7 +166,12 @@ in {
- 
-     linux_5_18 = throw "linux 5.18 was removed because it has reached its end of life upstream";
- 
--    linux_5_19 = throw "linux 5.19 was removed because it has reached its end of life upstream";
-+    linux_5_19 = callPackage ../os-specific/linux/kernel/linux-5.19.nix {
-+      kernelPatches = [
-+        kernelPatches.bridge_stp_helper
-+        kernelPatches.request_key_helper
-+      ];
-+    };
- 
-     linux_6_0 = callPackage ../os-specific/linux/kernel/linux-6.0.nix {
-       kernelPatches = [
-@@ -243,7 +248,7 @@ in {
-     linux_5_10_hardened = hardenedKernelFor kernels.linux_5_10 { };
-     linux_5_15_hardened = hardenedKernelFor kernels.linux_5_15 { };
-     linux_5_18_hardened = throw "linux 5.18 was removed because it has reached its end of life upstream";
--    linux_5_19_hardened = throw "linux 5.19 was removed because it has reached its end of life upstream";
-+    linux_5_19_hardened = hardenedKernelFor kernels.linux_5_19 { };
-     linux_6_0_hardened = hardenedKernelFor kernels.linux_6_0 { };
- 
-   }));
-@@ -528,7 +533,7 @@ in {
-     linux_5_16 = throw "linux 5.16 was removed because it reached its end of life upstream"; # Added 2022-04-23
-     linux_5_17 = throw "linux 5.17 was removed because it reached its end of life upstream"; # Added 2022-06-23
-     linux_5_18 = throw "linux 5.18 was removed because it reached its end of life upstream"; # Added 2022-09-17
--    linux_5_19 = throw "linux 5.19 was removed because it reached its end of life upstream"; # Added 2022-11-01
-+    linux_5_19 = recurseIntoAttrs (packagesFor kernels.linux_5_19);
-     linux_6_0 = recurseIntoAttrs (packagesFor kernels.linux_6_0);
-   };
- 
-@@ -568,7 +573,7 @@ in {
-     linux_5_10_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_5_10 { });
-     linux_5_15_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_5_15 { });
-     linux_5_18_hardened = throw "linux 5.18 was removed because it has reached its end of life upstream";
--    linux_5_19_hardened = throw "linux 5.19 was removed because it has reached its end of life upstream";
-+    linux_5_19_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_5_19 { });
-     linux_6_0_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_6_0 { });
- 
-     linux_zen = recurseIntoAttrs (packagesFor kernels.linux_zen);
diff --git a/pkgs/top-level/linux-kernels.nix b/pkgs/top-level/linux-kernels.nix
index efd71e23ce7..71c5750c75a 100644
--- a/pkgs/top-level/linux-kernels.nix
+++ b/pkgs/top-level/linux-kernels.nix
@@ -166,7 +166,12 @@ in {
 
     linux_5_18 = throw "linux 5.18 was removed because it has reached its end of life upstream";
 
-    linux_5_19 = throw "linux 5.19 was removed because it has reached its end of life upstream";
+    linux_5_19 = callPackage ../os-specific/linux/kernel/linux-5.19.nix {
+      kernelPatches = [
+        kernelPatches.bridge_stp_helper
+        kernelPatches.request_key_helper
+      ];
+    };
 
     linux_6_0 = callPackage ../os-specific/linux/kernel/linux-6.0.nix {
       kernelPatches = [
@@ -243,7 +248,7 @@ in {
     linux_5_10_hardened = hardenedKernelFor kernels.linux_5_10 { };
     linux_5_15_hardened = hardenedKernelFor kernels.linux_5_15 { };
     linux_5_18_hardened = throw "linux 5.18 was removed because it has reached its end of life upstream";
-    linux_5_19_hardened = throw "linux 5.19 was removed because it has reached its end of life upstream";
+    linux_5_19_hardened = hardenedKernelFor kernels.linux_5_19 { };
     linux_6_0_hardened = hardenedKernelFor kernels.linux_6_0 { };
 
   }));
@@ -528,7 +533,7 @@ in {
     linux_5_16 = throw "linux 5.16 was removed because it reached its end of life upstream"; # Added 2022-04-23
     linux_5_17 = throw "linux 5.17 was removed because it reached its end of life upstream"; # Added 2022-06-23
     linux_5_18 = throw "linux 5.18 was removed because it reached its end of life upstream"; # Added 2022-09-17
-    linux_5_19 = throw "linux 5.19 was removed because it reached its end of life upstream"; # Added 2022-11-01
+    linux_5_19 = recurseIntoAttrs (packagesFor kernels.linux_5_19);
     linux_6_0 = recurseIntoAttrs (packagesFor kernels.linux_6_0);
   };
 
@@ -568,7 +573,7 @@ in {
     linux_5_10_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_5_10 { });
     linux_5_15_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_5_15 { });
     linux_5_18_hardened = throw "linux 5.18 was removed because it has reached its end of life upstream";
-    linux_5_19_hardened = throw "linux 5.19 was removed because it has reached its end of life upstream";
+    linux_5_19_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_5_19 { });
     linux_6_0_hardened = recurseIntoAttrs (hardenedPackagesFor kernels.linux_6_0 { });
 
     linux_zen = recurseIntoAttrs (packagesFor kernels.linux_zen);
