diff --git a/pkgs/development/tools/parsing/tree-sitter/grammar.nix b/pkgs/development/tools/parsing/tree-sitter/grammar.nix
index d0265a584b379..96486c0c2b508 100644
--- a/pkgs/development/tools/parsing/tree-sitter/grammar.nix
+++ b/pkgs/development/tools/parsing/tree-sitter/grammar.nix
@@ -51,10 +51,10 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
-  # Auto strip cannot detect files missing extension.
-  fixupPhase = ''
+  # Strip failed on darwin: strip: error: symbols referenced by indirect symbol table entries that can't be stripped
+  fixupPhase = lib.optionalString stdenv.isLinux ''
     runHook preFixup
-    strip -s $out/parser
+    $STRIP $out/parser
     runHook postFixup
   '';
 }
