diff --git a/pkgs/applications/editors/vscode/update-vscodium.sh b/pkgs/applications/editors/vscode/update-vscodium.sh
index 08c776dd9..49a92b9fb 100755
--- a/pkgs/applications/editors/vscode/update-vscodium.sh
+++ b/pkgs/applications/editors/vscode/update-vscodium.sh
@@ -19,7 +19,6 @@ fi
 for i in \
     "x86_64-linux linux-x64 tar.gz" \
     "aarch64-linux linux-arm64 tar.gz" \
-    "armv7l-linux linux-armhf tar.gz" \
     "loongarch64-linux linux-loong64 tar.gz" \
     "aarch64-darwin darwin-arm64 zip"; do
     set -- $i
