diff --git a/pkgs/development/tools/yarn2nix-moretea/yarn2nix/internal/fixup_yarn_lock.js b/pkgs/development/tools/yarn2nix-moretea/yarn2nix/internal/fixup_yarn_lock.js
index 86e92f85..f77b60c9 100755
--- a/pkgs/development/tools/yarn2nix-moretea/yarn2nix/internal/fixup_yarn_lock.js
+++ b/pkgs/development/tools/yarn2nix-moretea/yarn2nix/internal/fixup_yarn_lock.js
@@ -34,7 +34,16 @@ readFile
 
       result.push(`  resolved "${fileName}#${shaOrRev}"`)
     } else {
-      result.push(line)
+      const arr2 = line.match(/^ {2}resolved "(.+)\/([^\/"]+)"$/)
+      if (arr2 !== null) {
+        const [_, url, shaOrRev] = arr2
+
+        const fileName = urlToName(url + '/' + shaOrRev)
+
+        result.push(`  resolved "${fileName}"`)
+      } else {
+        result.push(line)
+      }
     }
   })
   .on('close', () => {
