diff --git a/pkgs/os-specific/linux/zfs/default.nix b/pkgs/os-specific/linux/zfs/default.nix
index 5e92e63553d..19181d96d3f 100644
--- a/pkgs/os-specific/linux/zfs/default.nix
+++ b/pkgs/os-specific/linux/zfs/default.nix
@@ -16,7 +16,7 @@
 , enablePython ? true
 
 # for determining the latest compatible linuxPackages
-, linuxPackages_5_15 ? pkgs.linuxKernel.packages.linux_5_15
+, linuxPackages_5_19 ? pkgs.linuxKernel.packages.linux_5_19
 }:
 
 let
@@ -47,7 +47,8 @@ let
       name = "zfs-${configFile}-${version}${optionalString buildKernel "-${kernel.version}"}";
 
       src = fetchFromGitHub {
-        owner = "openzfs";
+        #owner = "openzfs";
+        owner = "tonyhutter";
         repo = "zfs";
         inherit rev sha256;
       };
@@ -216,28 +217,29 @@ in {
   # to be adapted
   zfsStable = common {
     # check the release notes for compatible kernels
-    kernelCompatible = kernel.kernelOlder "5.19";
-    latestCompatibleLinuxPackages = linuxPackages_5_15;
+    kernelCompatible = kernel.kernelOlder "5.20";
+    latestCompatibleLinuxPackages = linuxPackages_5_19;
 
     # this package should point to the latest release.
-    version = "2.1.5";
+    version = "2.1.6";
+    rev = "c5bdc6ca0e235302f4326fabd8eeeaa23399bdcb";
 
-    sha256 = "sha256-a9rmuPO8R8UfxdHvwjfFuYRGn97a1MPmLZRvr3l0swE=";
+    sha256 = "sha256-86s09kTr7nHuk9DGyGwWiCy47aRLWn1XbgMHyb7vFGc=";
   };
 
   zfsUnstable = common {
     # check the release notes for compatible kernels
-    kernelCompatible = kernel.kernelOlder "5.19";
-    latestCompatibleLinuxPackages = linuxPackages_5_15;
+    kernelCompatible = kernel.kernelOlder "5.20";
+    latestCompatibleLinuxPackages = linuxPackages_5_19;
 
     # this package should point to a version / git revision compatible with the latest kernel release
     # IMPORTANT: Always use a tagged release candidate or commits from the
     # zfs-<version>-staging branch, because this is tested by the OpenZFS
     # maintainers.
-    version = "2.1.5";
-    # rev = "0000000000000000000000000000000000000000";
+    version = "2.1.6";
+    rev = "c5bdc6ca0e235302f4326fabd8eeeaa23399bdcb";
 
-    sha256 = "sha256-a9rmuPO8R8UfxdHvwjfFuYRGn97a1MPmLZRvr3l0swE=";
+    sha256 = "sha256-86s09kTr7nHuk9DGyGwWiCy47aRLWn1XbgMHyb7vFGc=";
 
     isUnstable = true;
   };
